eval.go

```diff
@@ -407,23 +407,6 @@ func (d *indexData) gatherMatches(nextDoc uint32, mt matchTree, known map[matchT
 		}
 	})
 
-	// If there are content matches, trim all filename matches.
-	foundContentMatch := false
-	for _, c := range cands {
-		if !c.fileName {
-			foundContentMatch = true
-			break
-		}
-	}
-
-	res := cands[:0]
-	for _, c := range cands {
-		if !foundContentMatch || !c.fileName {
-			res = append(res, c)
-		}
-	}
-	cands = res
-
```

Comment:
We should update the doc string of gatherMatches

Reply:
Good catch!